### PR TITLE
feat(dsl): improvements in log analysis

### PR DESCRIPTION
**Pull request by doudou**
Depends on:
- [ ] https://github.com/rock-core/tools-roby/pull/210

This pull request mainly (greatly) improves object summary, and adds task argument as well
as event propagation info in Roby SQL logs, allowing to trace event calls and emissions
(previously only the actual emission was available in the log)

### CODE REVIEW

File: lib/syskit/log/roby_sql_index/index.rb [hunk 1]
```diff
@@ -139,19 +140,56 @@ def update_log_metadata(metadata, cycle)
...
                 # Add information about an emitted event
                 #
                 # @param [Roby::Event] ev
                 # @return [Integer] the record ID
-                def add_log_emitted_events(events)
-                    task_ids = add_log_tasks(events.map { |e| e.generator.task })
-                    records = events.zip(task_ids).map do |ev, task_id|
-                        { name: ev.symbol.to_s, time: ev.time, task_id: task_id }
+                def add_log_failed_emissions(records)
+                    records = records.map do |time, generator, _|
```
Comment (jhonasiv) [inline lib/syskit/log/roby_sql_index/index.rb:172]: rebuilder.plan.failed_emissions is composed by [generator, error].

File: lib/syskit/log/roby_sql_index/index.rb [hunk 2]
```diff
@@ -139,19 +140,56 @@ def update_log_metadata(metadata, cycle)
...
 
+                # @api private
+                #
+                # Add information about an event propagation (call or emission)
+                #
+                # @param [Array] records records, as stored in
+                #   {Roby::PlanRebuilder#propagated_events}
+                # @return [Integer] the record ID
+                def add_log_propagated_events(records)
+                    records = records.map do |time, is_forwarding, _, generator|
```
Comment (jhonasiv) [inline lib/syskit/log/roby_sql_index/index.rb:151]: rebuilder.plan.propagated_events is composed by [is_forwarding, events, generator]

**Comment by jhonasiv** [comment]
It seems easy to add time in these objects as both generator functions (`PlanRebuilder#generator_propagate_events` and `PlanRebuilder#generator_emit_failed`) receive time as an argument, but dont use it.

See the [source file](https://github.com/rock-core/tools-roby/blob/5a5ebac8c005816b297d62e7a76b9bd0ed0219ab/lib/roby/droby/plan_rebuilder.rb#L326).

**Comment by doudou** [comment]
> It seems easy to add time in these objects as both generator functions (`PlanRebuilder#generator_propagate_events` and `PlanRebuilder#generator_emit_failed`) receive time as an argument, but dont use it.
> 
> See the [source file](https://github.com/rock-core/tools-roby/blob/5a5ebac8c005816b297d62e7a76b9bd0ed0219ab/lib/roby/droby/plan_rebuilder.rb#L326).

Not sure what you meant by that ...

Anyways, there was a missing dependency on a Roby change.

**Comment by jhonasiv** [comment]
> > It seems easy to add time in these objects as both generator functions (`PlanRebuilder#generator_propagate_events` and `PlanRebuilder#generator_emit_failed`) receive time as an argument, but dont use it.
> > See the [source file](https://github.com/rock-core/tools-roby/blob/5a5ebac8c005816b297d62e7a76b9bd0ed0219ab/lib/roby/droby/plan_rebuilder.rb#L326).
> 
> Not sure what you meant by that ...
> 
> Anyways, there was a missing dependency on a Roby change.

What I meant was covered by https://github.com/rock-core/tools-roby/pull/210

File: lib/syskit/log/roby_sql_index/accessors.rb [hunk 3]
```diff
@@ -148,15 +207,37 @@ def method_missing(m, *args, **kw, &block)
...
+                        entity = @index.tasks.by_pk(task_id).where(model_id: id).one
+                        unless entity
+                            raise ArgumentError,
+                                  "no task with ID #{task_id} and model #{name}"
+                        end
+
+                        @tasks[task_id] = task_from_entity(entity)
                     end
 
+                    # Enumerate the
```
Comment (jhonasiv) [inline lib/syskit/log/roby_sql_index/accessors.rb:224]: Isolated comment

**Pull request merged**